README.md

```diff
@@ -1,2 +1,36 @@
 # py-template
-Template for a new python project
+This repository serves as a template for new python projects and a way to express best practices
+
+## Best Practices
+
+### User input
+* Always use `argparse` for command-line arguments
+* Assume the use of `yaml` for structured input files unless there are compelling reasons for something else
+
+### User output
+* Always use `logger` for status output
+* Carefully choose an output format for standard formats, considering the following in order of priority:
+    * CSV
+    * yaml
+    * json
+    * HDF5
+    * SQL
+
+### Modularity
+* All runnable scripts should include a block like:
```

Comment:
Maybe link this: https://stackoverflow.com/questions/419163/what-does-if-name-main-do to avoid Cargo-cult programming.